node_modules/oae-zencoder/tests/test-zencoder-api.js

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2016 Apereo Foundation (AF) Licensed under the
+ * Educational Community License, Version 2.0 (the 'License'); you may
+ * not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an 'AS IS'
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+var _ = require('underscore');
+var assert = require('assert');
+var fs = require('fs');
+var util = require('util');
+
+var RestAPI = require('oae-rest');
+var TestsUtil = require('oae-tests/lib/util');
+
+var ZencoderAPI = require('oae-zencoder/lib/api');
+var ZencoderDAO = require('oae-zencoder/lib/internal/dao');
+
+describe('Zencoder API', function() {
+
+    var globalAdminRestContext = null;
+    var camAdminRestContext = null;
+    var globalAdminOnTenantRestContext = null;
+
+    /*!
+     * Set up rest contexts before each test
+     */
+    beforeEach(function(callback) {
+        globalAdminRestContext = TestsUtil.createGlobalAdminRestContext();
+        camAdminRestContext = TestsUtil.createTenantAdminRestContext(global.oaeTests.tenants.cam.host);
+
+        // Log in the admin so his cookie jar is set up appropriately
+        RestAPI.Admin.loginOnTenant(globalAdminRestContext, 'localhost', null, function(err, ctx) {
+            assert.ok(!err);
+            globalAdminOnTenantRestContext = ctx;
+
+            RestAPI.User.getMe(globalAdminOnTenantRestContext, function(err, user) {
+                assert.ok(!err);
+                assert.ok(!user.anon);
+                callback();
+            });
+        });
+    });
+
+    /*!
+     * @return a stream to a video file
+     */
+    var _getVideoStream = function() {
+        return fs.createReadStream(__dirname + '/data/video.mp4');
+    };
+
+    describe('#encodingCompleted', function() {
+        it('verify a valid notification is processed correctly', function(callback) {
+            TestsUtil.generateTestUsers(globalAdminOnTenantRestContext, 1, function(err, users) {
+                assert.ok(!err);
+                var user = _.values(users)[0];
+                var restCtx = user.restContext;
+                // Create the content item that we'll test with
+                RestAPI.Content.createFile(restCtx, 'test video', null, 'private', _getVideoStream, null, null, null, function(err, content) {
+                    assert.ok(!err);
+                    ZencoderDAO.saveContentRevisionId('10', content.id, content.latestRevisionId, function(err) {
```

Comment:
Rather than doing this stuff manually, we should allow the preview-processing system to complete this for us -- a lot of this is because we run our unit tests as integration tests.

This wasn't happening for you before probably because previews are disabled by default in the unit tests. If you enable the preview processor, then you'll find that you can make use of the full preview processing subsystem. Please have a look at the tests for things like the link processor and pdf processor etc... and how they upload files to the API and then wait for them to be processed, then assert the results.